skylib/kustomize/kustomize.bzl

```diff
@@ -540,16 +540,24 @@ def _kubectl_impl(ctx):
         transitive_runfiles += [exe[DefaultInfo].default_runfiles for exe in trans_img_pushes]
 
     namespace = ctx.attr.namespace
+
+    variables = "--variable=NAMESPACE={namespace}".format(
+        namespace = namespace,
+    )
+    variables += " --variable=GIT_REVISION=\"$(git rev-parse HEAD)\""
```

Comment:
We should not do stamping of stamps in `.show` and `.apply` commands.
1. The git command will not work because the current directory is not the source directory.
2. We will be breaking the idempotency properties of `.show` and `apply` commands, which is no backward compatible behavior change.